src/python/pants/engine/process.py

```diff
@@ -65,6 +65,7 @@ class Process:
     use_nailgun: Digest
     execution_slot_variable: str | None
     cache_scope: ProcessCacheScope
+    reusable_input_digests: FrozenDict[str, Digest]
```

Comment:
Semantically, I think that calling these `immutable_input_digests` would more clearly specify their constraints, and better align with "append_only" (which is attempting to convey that caches must only grow). Otherwise, the first thought after seeing "reusable" is "don't I wanted everything to be reusable? i'll just put everything in there".